packages/typechain/src/parser/parseEvmType.ts

```diff
@@ -89,5 +89,9 @@ export function parseEvmType(rawType: string, components?: EvmSymbol[]): EvmType
     return { type: 'bytes', size: parseInt(match![1] || '1'), originalType: rawType }
   }
 
+  if (internalType?.startsWith('enum')) {
+    return parseEvmType('uint8')
+  }
```

Comment:
Enums aren't always encoded as `uint8`. From [Solidity ABI spec](https://solidity.readthedocs.io/en/latest/abi-spec.html#mapping-solidity-to-abi-types):
> | Solidity | ABI |
> |---|---|
> | `enum` | smallest uint type that is large enough to hold all values<br><br>For example, an enum of 255 values or less is mapped to uint8 and an enum of 256 values is mapped to uint16.

Changing this is currently being discussed (see https://github.com/ethereum/solidity/issues/10035 and [Solidity Language Design: Enums with max 256 values](https://groups.google.com/forum/#!msg/solidity-users/Y2DrM_dyBf4/e_jwFMATAwAJ)), but just FYI, very large enums that map `uint16` are currently possible.

Reply:
Thanks for this note, I was afraid that this assumption may break. I think it's gonna be just fine in our case as it's rather there just to make parsing happy...

Reply:
we can add a note in a comment there, just so no one is surprised or spends half a day resaarching if someone actually tries to use such a large enum :D

Reply:
Yup, I was planning on fixing it while working on TC today.